### DISCUSSION TITLE
Testing observable Value<T>

### DISCUSSION THREAD

**Discussion by thipokch** [arkivanov/Decompose#132] [Q&A] [answered]
How would you go about testing the `Value<Model>` abstraction without Reaktive? In other parts of the project, I've used [turbine](https://github.com/cashapp/turbine) to test my implementation of MviKotlin with coroutine extension. I could try testing the flow from the MviKotlin, but I'm not sure if that's a good idea. Thanks! ❤️

```kotlin
 override val models: Value<Model> = store.asValue().map {
        Model(
            a = it.a,
            b = it.b,
        )
    }

```

**Comment by arkivanov** [comment, marked as answer]
You can just verify the value: `some.models.value`. Please see an example: https://github.com/JetBrains/compose-jb/blob/620b50ae1205bccbc4982bcc5a6d4c86ef515edd/examples/todoapp/common/main/src/commonTest/kotlin/example/todo/common/main/integration/TodoMainTest.kt#L126

**Reply by thipokch** [reply]
Wow, thanks for the quick response. This works most of the time, however it doesn't work reliably when MviKotlin executes an async request that delays updates to model.

**Reply by arkivanov** [reply]
It's adviced do override schedulers/dispatchers in tests. For coroutines you can use `TestDispatcher` (or `UnconfinedDispatcher` for simple cases without delays).
